xml/System.Threading.Tasks/Task.xml

```diff
@@ -5445,11 +5445,25 @@ Task t Status: RanToCompletion, Result: 42
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- You can use `await Task.Yield();` in an asynchronous method to force the method to complete asynchronously. If there is a current synchronization context (<xref:System.Threading.SynchronizationContext> object), this will post the remainder of the method's execution back to that context. However, the context will decide how to prioritize this work relative to other work that may be pending. The synchronization context that is present on a UI thread in most UI environments will often prioritize work posted to the context higher than input and rendering work. For this reason, do not rely on `await Task.Yield();` to keep a UI responsive.  For more information, see the entry [Useful Abstractions Enabled with ContinueWith](https://blogs.msdn.com/b/pfxteam/archive/2008/07/23/8768673.aspx) in the Parallel Programming with .NET blog.  
+ You can use `await Task.Run();` in a method to force it to complete asynchronously. If there is a current synchronization context (<xref:System.Threading.SynchronizationContext> object), this will post the remainder of the method's execution back to that context. However, the context will decide how to prioritize this work relative to other work that may be pending. The synchronization context that is present on a UI thread in most UI environments will often prioritize work posted to the context higher than input and rendering work. For this reason, do not rely on `await Task.Yield();` or `await Task.Run();` to keep a UI responsive.  For more information, see the entry [Useful Abstractions Enabled with ContinueWith](http://blogs.msdn.com/b/pfxteam/archive/2008/07/23/8768673.aspx) in the Parallel Programming with .NET blog.  
+
+A good example of the usage would be a `Windows Forms` application where you would like to execute a lengthy operation, but at the same time release the thread so the UI would be accessible to user. This operation will get run at a later time (at which point, it still may run synchronously) on the current context.
+```cs
+async void Form_Load(object s, object e) 
+{ 
+    await Task.Run(() => LoadLocalConfiguration());  // unblock the UI before doing potentially lengthy operation
```

Comment:
Now, this example is not related to the containing article about [Task.Yield() method](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.yield?view=netframework-4.7.2). So, we've started with a `Task.Yield` example but ended up with a `Task.Run` alternative. Is there a short example of the `Task.Yield` use that *doesn't* have a better `Task.Run` alternative? If not, maybe the current **Remarks** section is good as it is now.

Reply:
Unit testing is a good use case for `Task.Yield`:  I often use it when mocking services to ensure that the code under test is exercised when it receives a Task that's not complete.  Mock example:

```csharp
// In a fake service:
public async Task<int> CountOrderFromDb)
{
   // simulate async operation reading a database
   // but use yield, not Task.Delay so it doesn't slow down text execution.
   await Task.Yield();
   return 5;
}
```